default.nix

```diff
@@ -1,22 +1,23 @@
 let
+  # this overlay allows us to use a specified version of the rust toolchain
   rustOverlay =
     import (builtins.fetchGit {
       url = "https://github.com/mozilla/nixpkgs-mozilla.git";
       rev = "f233fdc4ff6ba2ffeb1e3e3cd6d63bb1297d6996";
     });
+
+  # pinned version of nix packages
```

Comment:
An explanation as to why we are using this version (not the newest at time of writing) would be useful (points about compatibility with Ubuntu LTS; does that even work now?)